### DISCUSSION TITLE
How to reset person account credentials via REST interface?

### DISCUSSION THREAD

**Discussion by TimoKramer** [kanidm/kanidm#2537] [Q&A] [answered]
Hi,
I am reading [about resetting the users' credentials ](https://kanidm.github.io/kanidm/stable/accounts/authentication.html#resetting-person-account-credentials) and wonder how to do that via rest. Is there some docs about the rest interface details? The generated swagger is missing a lot of details.

Which endpoint can I use for resetting the users' credentials?
Thanks, Timo

**Comment by Firstyear** [comment, marked as answer]
You can't reset a users credentials on their behalf. You generate a credential reset token, and provide that to the user so they can then perform an interactive/once off reset of their own credentials. A credential reset token may only be used once. To create a reset token the person or service-account calling the api must have write access to the account. This is limited to the account-itself, or members of the groups `idm_people_admins`, `idm_service_desk`, or `idm_people_on_boarding`. If you plan to use a service-account to do this to create and sign-up new users, you should use the `idm_people_on_boarding` group.

This can be triggered by `kanidm person credential create-reset-token` or the API endpoint `/v1/person/:id/_credential/_update_intent`, where ":id" is a unique identifier (uuid, name or spn) of the account to generate the reset token for.
